datatypes/src/main/java/com/fasterxml/jackson/datatype/jdk8/StreamSerializer.java

```diff
@@ -0,0 +1,67 @@
+package com.fasterxml.jackson.datatype.jdk8;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.ser.ContextualSerializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+/**
+ * Common typed stream serializer
+ *
+ * @param <T> type of the stream elements
+ */
+public class StreamSerializer<T> extends StdSerializer<Stream<T>> implements ContextualSerializer {
+    /**
+     * Stream elements type (matching T)
+     */
+    private final JavaType elemType;
+    /**
+     * element specific serializer, if any
+     */
+    private transient JsonSerializer<T> elemSerializer = null;
+
+    /**
+     * Constructor
+     *
+     * @param streamType Stream type
+     * @param elemType   Stream elements type (matching T)
+     */
+    public StreamSerializer(JavaType streamType, JavaType elemType) {
+        super(streamType);
+        this.elemType = elemType;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public JsonSerializer<?> createContextual(SerializerProvider provider, BeanProperty property) throws JsonMappingException {
+        if (!elemType.hasRawClass(Object.class) && (provider.isEnabled(MapperFeature.USE_STATIC_TYPING) || elemType.isFinal())) {
+            elemSerializer = (JsonSerializer<T>) provider.findPrimaryPropertySerializer(elemType, property);
```

Comment:
Quick note: modifying instance is not allowed, `createContextual()` MUST create and return a new instance -- otherwise cached "blueprint" instance is modified.
So there needs to be bit more plumbing to create new instance (define abstract factory method here, let sub-classes implement).